src/algebra/monoid_algebra.lean

```diff
@@ -373,6 +377,81 @@ end
 
 end misc_theorems
 
+/-! #### Non-unital, non-associative algebra structure -/
+section non_unital_non_assoc_algebra
+
+variables (k) [semiring k] [has_mul G]
+
+instance is_scalar_tower_self : is_scalar_tower k (monoid_algebra k G) (monoid_algebra k G) :=
```

Comment:
I think this might hold more generally for:
```suggestion
instance is_scalar_tower_self [is_scalar_tower R k k] : is_scalar_tower R (monoid_algebra k G) (monoid_algebra k G) :=
```

Reply:
Yeah, this works:
```
variables {R : Type*} (k) [semiring R] [semiring k] [module R k] [has_mul G]

instance is_scalar_tower_self [is_scalar_tower R k k] :
  is_scalar_tower R (monoid_algebra k G) (monoid_algebra k G) :=
⟨λ t a b, begin
  ext m,
  simp [mul_apply, finsupp.smul_sum, smul_ite, smul_mul_assoc, sum_smul_index'],
end⟩
```

Reply:
This actually holds even for `distrib_mul_action R k` I think, but first `finsupp.smul_sum` needs its typeclasses relaxing

Reply:
Done, and can confirm #8028 did suffice. Thanks!